tests/main.rs

```diff
@@ -112,26 +140,70 @@ where
         self
     }
 
-    #[cfg(feature = "tls")]
-    fn with_server_tls(mut self) -> Self {
-        let cert = rcgen::generate_simple_self_signed(vec!["localhost".to_string()]).unwrap();
-
-        self.server_tls = Some(std::sync::Arc::new(
-            ServerConfig::builder()
-                .with_safe_defaults()
-                .with_no_client_auth()
-                .with_single_cert(
-                    vec![Certificate(cert.serialize_der().unwrap())],
-                    PrivateKey(cert.get_key_pair().serialize_der()),
-                )
-                .unwrap(),
-        ));
+    #[cfg(all(feature = "tls", unix))]
+    fn with_tls(mut self, client: bool, server: bool, use_client_certs: bool) -> Self {
+        use std::fs::File;
+        use std::io::Write;
 
-        self
-    }
+        let mut client_cert_der = None;
+
+        if use_client_certs {
+            let (client_cert, client_pkey) = mk_client_cert().unwrap();
+
+            client_cert_der = Some(Certificate(client_cert.to_der().unwrap()));
+
+            // Set up client cert der12 file.
+            let client_cert_pkcs12_file = Arc::new(tempfile::NamedTempFile::new().unwrap());
+            self.client_cert_pkcs12_file = Some(Arc::clone(&client_cert_pkcs12_file));
+
+            let pkcs12 = Pkcs12::builder()
+                .build("password", "friendly_name", &client_pkey, &client_cert)
+                .unwrap();
+            let der = pkcs12.to_der().unwrap();
+
+            let mut f = File::create(&*client_cert_pkcs12_file).unwrap();
+            f.write(&der).unwrap();
+            f.flush().unwrap();
+        }
+
+        if server {
+            let cert = generate_simple_self_signed(vec!["localhost".to_string()]).unwrap();
+
+            let builder = ServerConfig::builder().with_safe_defaults();
+
+            let builder = if use_client_certs {
```

Comment:
```suggestion
            let builder = if let Some(client_cert_der) = client_cert_der {
```
That way you can avoid the `unwrap` below.